boards: st: nucleo_h563zi: fix PLL clocks

mariopaja opened the pull request:
This change aligns max allowed SPI clock with the datasheet.

SPI1 on H5xx is using PLL1Q for their clock selection
https://github.com/zephyrproject-rtos/zephyr/blob/64a1477976b62606e8844a4bc8de880117a1cc47/dts/arm/st/h5/stm32h5.dtsi#L510

Based on [STM32H562/3xx Datasheet](https://www.st.com/resource/ja/datasheet/stm32h563ri.pdf), the  maximum allowed clock freq for SPI is 125MHz.
<img width="873" height="87" alt="f_spi" src="https://github.com/user-attachments/assets/c5118d24-97d8-4936-9a01-9f28aadd4dd4" />

By changing PLL1Q div to 4, SPI clock drops to 120MHz (just below the max allowed frequency)

The new proposed PLL clocks:
1. PLL1 will be used by SPI
2. PLL2 for FDCAN (as discussed) because clock freq should remain as is
3. PLL3 will be used in the _samples/i2s/output_ for the SAI

<img width="663" height="678" alt="h563_pll" src="https://github.com/user-attachments/assets/7c545f67-9490-4311-9c3b-29cd7a76d358" />

mariopaja commented:
This issue https://github.com/zephyrproject-rtos/zephyr/issues/91103 could have been related to this. (just speculating not tested)

erwango commented:
Thanks for the fix @mariopaja.
I can see that FDCAN clock is also using PLL1_Q output. Would you mind checking this change won't have negative impact on that front?

gautierg-st reviewed (changes requested):
Thanks. Though could you also update `stm32h562.dtsi` while you're at it ?
Both sai nodes indicates `<&rcc STM32_SRC_PLL2_P SAI1_SEL(0)>` but it should be `<&rcc STM32_SRC_PLL1_P SAI1_SEL(0)>`

mariopaja commented:
> Thanks. Though could you also update `stm32h562.dtsi` while you're at it ? Both sai nodes indicates `<&rcc STM32_SRC_PLL2_P SAI1_SEL(0)>` but it should be `<&rcc STM32_SRC_PLL1_P SAI1_SEL(0)>`

I updated it, but maybe the sai nodes of all platforms should be updated to the default value. Atm I dont know if they use the default value, but it would have to be checked and updated accordingly?

erwango commented on dts/arm/st/h5/stm32h562.dtsi:
```diff
@@ -128,7 +128,7 @@
 			#size-cells = <0>;
 			reg = <0x40015404 0x20>;
 			clocks = <&rcc STM32_CLOCK(APB2, 21)>,
-				 <&rcc STM32_SRC_PLL2_P SAI1_SEL(0)>;
```
.dtsi change should be done in a dedicated commit

mariopaja replied:
Should I leave it as is and check all SAI nodes in another PR?

erwango replied:
As you wish, but ok for me to keep in current PR.

erwango reviewed (changes requested):
Otherwise LGTM

mariopaja committed:
boards: st: nucleo_h563zi: fix PLL clocks

These changes fix PLL clocks of nucleo_h563zi board

Signed-off-by: Mario Paja <mariopaja@hotmail.com>

mariopaja committed:
dts: st: h5: set sai default clock configuration

This change sets SAI clock to its default configuration

Signed-off-by: Mario Paja <mariopaja@hotmail.com>
